fission/src/ui/modals/configuring/SettingsModal.tsx

```diff
@@ -1,23 +1,14 @@
 import React, { useState } from "react"
 import { useModalControlContext } from "@/ui/ModalContext"
 import Modal, { ModalPropsImpl } from "@/components/Modal"
-import { FaGear } from "react-icons/fa6"
 import Label, { LabelSize } from "@/components/Label"
-import Dropdown from "@/components/Dropdown"
 import Button from "@/components/Button"
 import Slider from "@/components/Slider"
 import Checkbox from "@/components/Checkbox"
 import PreferencesSystem from "@/systems/preferences/PreferencesSystem"
 import { SceneOverlayEvent, SceneOverlayEventKey } from "@/ui/components/SceneOverlayEvents"
-
-const moveElementToTop = (arr: string[], element: string | undefined) => {
-    if (element == undefined) {
-        return arr
-    }
-
-    arr = arr.includes(element) ? [element, ...arr.filter(item => item !== element)] : arr
-    return arr
-}
+import Dropdown from "@/ui/components/Dropdown"
+import { SynthesisIcons } from "@/ui/components/StyledComponents"
 
 const screenModeOptions = ["Windowed", "Fullscreen"]
 const qualitySettingsOptions = ["Low", "Medium", "High", "Ultra"]
```

Comment:
Graydon's rolling in his metaphorical grave right now

Reply:
@azaleacolburn is this better?

```ts
enum QualitySettingsCharacter {
    L,
    O,
    W,
    M,
    E,
    D,
    I,
    U,
    H,
    G,
}

const getSetting = (settings: QualitySettingsCharacter[]) => {
    if (
        settings[0] == QualitySettingsCharacter.L &&
        settings[1] == QualitySettingsCharacter.O &&
        settings[2] == QualitySettingsCharacter.W
    )
        return "Low"
    else if (
        settings[0] == QualitySettingsCharacter.M &&
        settings[1] == QualitySettingsCharacter.E &&
        settings[2] == QualitySettingsCharacter.D &&
        settings[3] == QualitySettingsCharacter.I &&
        settings[4] == QualitySettingsCharacter.U &&
        settings[5] == QualitySettingsCharacter.M
    )
        return "Medium"
    else if (
        settings[0] == QualitySettingsCharacter.H &&
        settings[1] == QualitySettingsCharacter.I &&
        settings[2] == QualitySettingsCharacter.G &&
        settings[3] == QualitySettingsCharacter.H
    )
        return "High"
}
```

(this is a joke btw)

Reply:
You're a genius, think about it, the JIT compiler will read the enums faster, since each enum name is only one character long

Reply:
@azaleacolburn serious question time: Is there an advantage of using an enum instead of 
`type QualitySetting = "Low" | "Medium" | "High"`

Reply:
Yes, but not a _ton_
- Enums can be iterated over
- Enums provide clear intent about what is happening (enums are for a list of valid options / variants, while optional types were introduced as a way of allowing variant types. The idea is that optional types should be used for type variation, rather than a list of variant values.) This is the important one for me.
- Enums are slightly faster at runtime since they're just numbers.
- Maybe my lsp is broken, but i get autocomplete on enums and not on enum-like string type variants.
- In every place where a variant is directly used, it's clearer about what the variant means with an enum, than if just a string literal is used. the string literal could mean anything, while the enum is clearly a variant of an some option. Think about the difference between seeing the literal `SimType.PWM` passed in compared to just `"PWM"`, to me, if I'm unfamiliar with the code, one is clearer.

Cons:
- Enum syntax is uglier